pii_recognizer/pii_recognizer.py

```diff
@@ -0,0 +1,590 @@
+    :returns: The list of tokens with the identified entities.
+
+    """
+    tokens = []
+    # sort by start index
+    results = sorted(st_analyze_results, key=lambda x: x.start)
+    for i, res in enumerate(results):
+        if i == 0:
+            tokens.append(text[: res.start])
+
+        # append entity text and entity type
+        tokens.append((text[res.start : res.end], res.entity_type))
+
+        # if another entity coming i.e. we're not at the last results element,
+        # add text up to next entity
+        if i != len(results) - 1:
+            tokens.append(text[res.end : results[i + 1].start])
+        # if no more entities coming, add all remaining text
+        else:
+            tokens.append(text[res.end :])
+    return tokens
+
+
+class CustomEncoder(JSONEncoder):
+    """Custom encoder for JSONEncoder.
+    :param JSONEncoder: The JSONEncoder.
+    :returns: The JSONEncoder.
+    """
+
+    def default(self, o):
+        return o.__dict__
+
+
+def process(text: str, model: AnalyzerEngine):
+    """
+    Process the text of str using the model.
+    :param txt: Text to process
+    :param model: Model to use for processing
+    :returns: The anonymized text, the html string and the stats report.
+    """
+    analyzer = model
+    results = analyzer.analyze(
+        text=text,
+        language="en",
+        entities=get_supported_entities(),
+        return_decision_process=True,
+    )
+    anonymized_text = anonymize(text, results)
+
+    annotated_tokens = annotate(text, results, get_supported_entities())
+    html = get_annotated_html(*annotated_tokens)
+    backslash_char = "\\"
+
+    html_str = f"<p>{html.replace('{backslash_char}n', '<br>')}</p>"
+
+    stats = results
+
+    return anonymized_text, html_str, stats
+
+
+def pii_recognize(
+    context: mlrun.MLClientCtx,
+    input_path: str,
+    output_path: str,
+    output_suffix: str,
+    html_key: str,
+    model: str = "whole",
+) -> str:
+    """
+    Recognize PII in text.
+    :param context: The MLRun context.
+    :param input_path: The input path to the artifact.
+    :param output_path: The output path to store the anonymized text.
+    :param output_suffix: The surfix of output key for the anonymized text. for example if the input file is pii.txt, the output key will be pii_anonymized.txt.
+    :param html_key: The html key for the artifact.
+    :param model: The model to use. Can be "spacy", "flair", "pattern" or "whole".
+    :returns: output_path
+
+    """
+    if not output_path:
+        output_path = input_path + "/output/"
+    analyzer = analyzer_engine(model)
+    txt_file_paths, txt_file_names = get_text_files(input_path)
+    html_index = "<html><head><title>Highlighted Pii Entities</title></head><body><h1>Highlighted Pii Entities</h1><ul>"
+    html_content = ""
+    rpt_json = {}
+    for file_path, file_name in zip(txt_file_paths, txt_file_names):
+        text = mlrun.get_dataitem(file_path).get().decode("utf-8")
```

Comment:
you should have the file already locally, no need to `get_data_item`. you can get a directory of files or a single file as input, all should be handled automatically by MLRun.

Reply:
Cool. I will try this out.